src/shared/fs/templateRegistry.ts

```diff
@@ -108,31 +102,37 @@ export class AsyncCloudFormationTemplateRegistry {
             return this.instance
         }
 
-        // Show user a message indicating setup is in progress
-        if (this.setupProgressMessage === undefined) {
-            this.setupProgressMessage = vscode.window.withProgress(
-                {
-                    location: vscode.ProgressLocation.Notification,
-                    title: localize(
-                        'AWS.codelens.waitingForTemplateRegistry',
-                        'Scanning CloudFormation templates... (except paths configured in [search.exclude](command:workbench.action.openSettings?"@id:search.exclude"))'
-                    ),
-                    cancellable: true,
-                },
-                async (progress, token) => {
-                    token.onCancellationRequested(() => {
-                        // Allows for new message to be created if templateRegistry variable attempted to be used again
-                        this.setupProgressMessage = undefined
-                    })
-                    getLogger().debug('cfn: getInstance() requested, still initializing')
-                    while (!this.isSetup) {
-                        await sleep(2000)
+        const config = SamCliSettings.instance
+        if (config.get('enableCodeLenses', false)) {
+            this.asyncSetupFunc(this.instance).then(() => {
+                this.isSetup = true
+                return this.instance
+            })
```

Comment:
Isn't the PerfLog still useful to have? It's not only for startup performance, it's for measuring any kind of performance.

Reply:
I'll see if I can worm it in there. It's a little trickier now that this builds on demand and not on the single constructor call.